src/pages/index.js

```diff
@@ -1,80 +1,170 @@
-const locationPopup = new PopupWithImage("#location-modal");
+const popups = {
+  cardPopup: new PopupWithForm("#add-card-modal", renderNewCard),
+  profilePopup: new PopupWithForm("#edit-profile-modal", editProfile),
+  deleteCardPopup: new PopupWithForm("#delete-card-modal", deleteCard),
+  profilePicPopup: new PopupWithForm("#profile-pic-modal", editProfilePic),
+  locationPopup: new PopupWithImage("#location-modal"),
+};
 
-const userInfo = new UserInfo({
-  nameSelector: ".profile__name",
-  descriptionSelector: ".profile__subtitle",
-});
-const cardSection = new Section(
-  { items: initialCards, renderer: createCard },
-  ".locations"
+const userInfo = new UserInfo(
+  {
+    nameSelector: ".profile__name",
+    descriptionSelector: ".profile__subtitle",
+  },
+  profileImage
 );
+const cardSection = new Section({ renderer: createCard }, ".locations");
 
 // validation class
-const editFormValidation = new FormValidator(validationObj, editForm);
-const cardFormValidation = new FormValidator(validationObj, addCardForm);
+const formValidators = {
+  editFormValidation: new FormValidator(validationObj, editForm),
+  cardFormValidation: new FormValidator(validationObj, addCardForm),
+  picFormValidation: new FormValidator(validationObj, profilePicForm),
+};
 
 // initail setup
-cardPopup.setEventListeners();
-profilePopup.setEventListeners();
-locationPopup.setEventListeners();
+for (let popup in popups) {
+  popups[popup].setEventListeners();
+}
 
-cardSection.renderItems();
+for (let form in formValidators) {
+  formValidators[form].enableValidation();
+}
 
-editFormValidation.enableValidation();
-cardFormValidation.enableValidation();
+api
+  .getAll()
+  .then(([userData, cardsData]) => {
+    userInfo.setUserInfo(userData);
+    userInfo.setUserImage(userData);
+    popups.profilePopup.setInputs(userInfo.getUserInfo());
+    cardSection.renderItems(cardsData);
+  })
+  .catch((err) => console.error(err));
 
 // edits the user profile based off form inputs
-function editProfile(e, { name, description }) {
-  userInfo.setUserInfo({ name, description });
-  profilePopup.closeModal();
+function editProfile({ name, description }) {
+  popups.profilePopup.renderLoading(true);
+  api
+    .updateUserInfo({ name, description })
+    .then(() => {
+      userInfo.setUserInfo({ name, description });
+      popups.profilePopup.closeModal();
+      popups.profilePopup.renderLoading();
+    })
+    .catch((err) => console.error(err));
+}
+
+function editProfilePic({ avatar }) {
+  popups.profilePicPopup.renderLoading(true);
+  api
+    .updateUserImage({ avatar })
+    .then(() => {
+      userInfo.setUserImage({ avatar });
+      popups.profilePicPopup.closeModal();
+      popups.profilePicPopup.renderLoading();
+      popups.profilePicPopup.form.reset();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => formValidators.picFormValidation.toggleSubmitButton());
```

Comment:
Needs correction: please toggle the button in then block (only after the successful api response, there is no need to toggle the button in the case of an error)